### PR TITLE
Query database optimization documentation

### PR DESCRIPTION
**Pull request by DL6ER**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/contributing/).

---

**What does this PR aim to accomplish?:**

Documentation update following https://github.com/pi-hole/FTL/pull/1255

### CODE REVIEW

**Comment by yubiuser** [comment]
> The long-term database contains three tables:

Actually, quite some more now ;-) (Also messages, network, nework_addresses, but this is for another PR..)

I would add to "Query table" "(`queries_storage`)" and adjust the the type of the columns (client is not text anymore).

File: docs/database/ftl.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ Label | Type | Allowed to by empty | Content
 `forward` | text | Yes | Forward destination used for this query (only set if `status == 2`)
 `additional_info` | blob | Yes | Data-dependent content, see below
 
+The `queries` `VIEW` is dynamically generated from the data actually stored in the tables `queries_storage` and the linking tables `domain_by_id`, `client_by_id`, `forward_by_id`, and `addinfo_by_id` (see below). The table `queries_storage` will contains integer IDs pointing to the respective entries of the linking tables to save space and make searching the database faster. If you haven't upgraded for some time, the table may still contain strings instead of integer IDs.
+
 #### Data-dependent `additional_info` field
```
Comment (yubiuser) [inline docs/database/ftl.md:56]: This can go to `addinfo_by_id` now.

**Comment by DL6ER** [comment]
> I would add to "Query table" "(queries_storage)" and adjust the the type of the columns (client is not text anymore).

I disagree on this part. It doesn't matter for consumers how the data is really stored in the database. They are supposed to query the database through he VIEW and not the raw data storage because the latter needs severe amount of post-processing that may be too complicated for SQL beginners. The column type `TEXT` is still accurate as  the type returned by the `VIEW`.

File: docs/database/ftl.md (hunk 1 above)
Comment (DL6ER) [reply docs/database/ftl.md:56]: I disagree for the same reasons as outlined [here](https://github.com/pi-hole/docs/pull/645#issuecomment-1021757398). Users are supposed to interact with the database on a sufficiently high level to not cause unnecessary work for them and open the potential for potential (and unnecessary) harm when they interact with the database on the lower levels (while this isn't necessary).

Comment (yubiuser) [reply docs/database/ftl.md:56]: Ok, convinced.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**